docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch

```diff
@@ -0,0 +1,145 @@
+From ddf20fca1e5d3f5669909e825b6b20332887d006 Mon Sep 17 00:00:00 2001
+From: Quentin Boyer <quentin+dev@familleboyer.net>
+Date: Wed, 11 Sep 2024 00:05:47 +0200
+Subject: [PATCH] nixos-render-docs: Support gfm-style admonitions
+
+---
+ nixos_render_docs/md.py  | 35 +++++++++++++++++++++++++++++++++++
+ tests/sample_md.py       |  3 +++
+ tests/test_asciidoc.py   |  6 ++++++
+ tests/test_commonmark.py |  3 +++
+ tests/test_html.py       |  4 ++++
+ tests/test_manpage.py    |  6 ++++++
+ 6 files changed, 57 insertions(+)
+
+diff --git a/nixos_render_docs/md.py b/nixos_render_docs/md.py
+index 894daf9ca9c7..cde874bc5637 100644
+--- a/nixos_render_docs/md.py
++++ b/nixos_render_docs/md.py
+@@ -341,6 +341,40 @@ def _parse_blockattrs(info: str) -> Optional[tuple[AttrBlockKind, Optional[str],
+         return ('figure', id, classes)
+     return None
+ 
++_ALERT_PATTERN = re.compile(r"^\[\!(TIP|NOTE|IMPORTANT|WARNING|CAUTION)\]", re.IGNORECASE)
++
++def _gfm_alert(md: markdown_it.MarkdownIt):
++    def gfm_alert_to_adm(state: markdown_it.rules_core.StateCore):
++        tokens = state.tokens
++        for (i, token) in enumerate(tokens):
++            if token.type == "blockquote_open":
++                open = tokens[i]
++                start_index = i
++                while tokens[i].type != "blockquote_close" and i < len(tokens):
++                    i += 1
++                close = tokens[i]
++                end_index = i
++
++                first_content = next(
++                    (t for t in tokens[start_index:end_index + 1] if t.type == "inline"),
++                    None,
++                )
++                if first_content is None:
++                    continue
++
++                if (m := _ALERT_PATTERN.match(first_content.content)) is None:
++                    continue
```

Comment:
I like the idea of splitting this into another function, but it isn't that important.

Reply:
What part would you want in another function?

Reply:
I was originally thinking we could be inspired by [parser.ts](https://github.com/ByPikod/markdown-it-github-alert/blob/f17613451396d7aa43cf6e8a3a81a4663ff1f27b/src/parser.ts#L13), which exports a "matcher" function returning either `null` or the matched type.

However, I think it is fine to keep this inline as it makes removing the matched substring easier.